### ISSUE TITLE
Utility Button is Not Properly Highlighted in iOS

### ISSUE THREAD

**Issue by foxannefoxanne** [cerner/terra-framework#736]
# Bug Report

## Description
This appears to be the same issue as: https://github.com/cerner/terra-framework/issues/277 but scoped to iOS

When using Terra Application layout in iOS view, clicking the utility button does not change focus to the utility menu. 

## Steps to Reproduce
1. Go to https://engineering.cerner.com/terra-framework/#/home on iOS with Screen Reader (VoiceOver) turned on
2. Select utility menu in small (hamburger) layout
3. Note that focus is not shifted, instead you are stuck on 'utility button - button' 

## Expected Behavior
Utility button will switch focus to utility menu, as it does in other layouts. 

## Environment
Although reproducible on https://engineering.cerner.com/terra-framework/#/home/terra-ui/index, here's information on our system
* Component Name and Version:  terra-application-layout, 4.8.0
* Browser Name and Version: Safari 12
* Node/npm Version: lts/carbon in our app
*  Webpack Version: 3.11.0
* Operating System and version (desktop or mobile): iOS 12

**Comment by pranav300** [comment]
1. If `Utility Button is Not Properly Highlighted in iOS` then that could be because, `By default, tab-access is disabled in Safari. To enable it, check "Preferences > Advanced > Press tab to highlight each item on a page".`

3. if `Note that focus is not shifted, instead you are stuck on 'utility button - button'` it is not reproducible in the desktop browser and can only be seen on the mobile browser and reason for that might be the double tap to click the button.

**Comment by bjankord** [comment]
From my testing so far, it looks like this may only be an issue in Safari on iOS when VoiceOver is enabled. When testing with VoiceOver in Safari on desktop, the focus was shifted to the modal dialog correctly when selecting the utility menu button in the side nav in narrow viewports.

**Comment by bjankord** [comment]
Found a related issue in another project: https://github.com/humaan/Modaal/issues/99

The comments on that thread are interesting and may shed some light on some of the odd behavior we've been seeing with VoiceOver on iOS.
I'm going to look into setting a timeout to see if that can help trigger the focus shift correctly to account for VoiceOver being slow to pick up that new DOM nodes are added/updated. Seems like the accessibility tree isn't getting populated fast enough or something.

**Comment by bjankord** [comment]
I'm working on a fix for this and should have a PR out for it shortly.
In it, I'm planning to add back functionality similar to focus-trap-react that would focus on the first interactive DOM element in the modal when it is opened via the use of [tabbable](https://www.npmjs.com/package/tabbable). This seems to allow VoiceOver on iOS to correctly shift focus into the modal dialog when it is opened.
